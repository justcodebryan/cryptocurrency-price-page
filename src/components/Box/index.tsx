import classNames from 'classnames'
import { DetailedHTMLProps, HTMLAttributes, PropsWithChildren } from 'react'

import styles from './styles.module.scss'

type BoxProps = PropsWithChildren<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>>

const Box = ({ children, className, ...rest }: BoxProps) => {
  const internalClass = classNames([styles['box'], className])
  return (
    <div className={internalClass} {...rest}>
      {children}
    </div>
  )
}

export default Box
