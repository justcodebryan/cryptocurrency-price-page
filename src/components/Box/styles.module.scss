.box {
  width: calc(100% - 24px * 2);
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}
