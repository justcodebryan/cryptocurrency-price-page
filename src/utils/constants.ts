import { NoopFn } from '@/types'

export const NOOP: NoopFn = () => void 0

export const DEFAULT_MODAL_CONTAINER_ID = 'modal-container'

export const DEFAULT_POLLING_INTERVAL = 10 * 1000
