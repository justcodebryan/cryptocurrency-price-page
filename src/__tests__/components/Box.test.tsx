import Box from '@/components/Box'
import { toJSON } from '@/utils/test-utils'
import renderer from 'react-test-renderer'

describe('Box Test Suit', () => {
  test('Box Snapshot', () => {
    const component = renderer.create(<Box />)
    const tree = toJSON(component)
    expect(tree).toMatchSnapshot()
  })
})
