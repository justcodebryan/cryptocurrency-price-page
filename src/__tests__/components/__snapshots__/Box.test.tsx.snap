// Vitest Snapshot v1

exports[`Box Test Suit > Box Snapshot 1`] = `
<div
  className="_box_6c0536"
/>
`;
